polaris-service/src/main/java/org/apache/polaris/service/persistence/cache/EntityCacheFactory.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.persistence.cache;
+
+import jakarta.inject.Inject;
+import org.apache.polaris.core.context.CallContext;
+import org.apache.polaris.core.context.RealmScope;
+import org.apache.polaris.core.persistence.PolarisMetaStoreManager;
+import org.apache.polaris.core.persistence.cache.EntityCache;
+import org.glassfish.hk2.api.Factory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EntityCacheFactory implements Factory<EntityCache> {
+  private static Logger LOGGER = LoggerFactory.getLogger(EntityCacheFactory.class);
+  @Inject PolarisMetaStoreManager metaStoreManager;
+
+  @RealmScope
```

Comment:
Again I think it's much easier to turn this bean into application-scoped and have it hold a map of entity caches keyed by realm id, like `RealmEntityManagerFactory`.